src/ModelObjects/component.rs

```diff
@@ -255,6 +255,218 @@ impl Component {
         self.output_edges = Some(o_edges);
         self.input_edges = Some(i_edges);
     }
+
+    /// Function for reducing the clocks found on the component.
+    /// Unused clocks and "duplicate" clocks (clocks that are never reset)
+    /// and then remove them.
+    pub fn reduce_clocks(&mut self, redundant_clocks: &Vec<RedundantClock>) {
+        for clock in redundant_clocks {
+            match &clock.reason {
+                ClockReductionReason::Duplicate(global) => self.replace_clock(&clock, global),
+                ClockReductionReason::Unused => self.remove_clock(&clock.updates),
+            }
+
+            let clock_val = *self.declarations.clocks.get(clock.clock.as_str()).unwrap();
```

Comment:
Is this safe? What if it do not get any clocks? (Error handling a unwrap)

Reply:
So, the reason for using `unwrap()` without error-handling is because in `redundant_clocks`, we only loop over clocks that have been found on `self`, meaning we know that it exists.
This is the same idea we have the other places we use `unwrap()` without error-handling (we have filtered it beforehand, meaning we know that it is `Some(<T>)`).
I can implement a `.expect()` with a panic-message if you like?

Reply:
I would like that, but if you can find the place where you get the clocks for the first time you can add the .expect() there

Reply:
done